Symfony/CS/Fixer/Contrib/MethodArgumentDefaultValueFixer.php

```diff
@@ -0,0 +1,237 @@
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'In method arguments there must not be arguments with default values before non-default ones.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+        for ($i = 0, $l = $tokens->count(); $i < $l; ++$i) {
+            if ($tokens[$i]->isGivenKind(T_FUNCTION)) {
+                $this->fixFunctionDefinition($tokens, $i);
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function fixFunctionDefinition(Tokens $tokens, $index)
+    {
+        $examinedIndex = $tokens->getNextTokenOfKind($index, $this->argumentBoundaryTokens);
+        $lastNonDefaultArgumentIndex = $this->getLastNonDefaultArgumentIndex($tokens, $index);
+
+        while (
+            $examinedIndex < $lastNonDefaultArgumentIndex &&
+            $this->hasDefaultValueAfterIndex($tokens, $examinedIndex)
+        ) {
+            $nextRelevantIndex = $tokens->getNextTokenOfKind($examinedIndex, $this->variableOrTerminatorTokens);
+
+            if (!$tokens[$nextRelevantIndex]->isGivenKind(T_VARIABLE)) {
+                break;
+            }
+
+            if (
+                $this->isDefaultArgumentAfterIndex($tokens, $nextRelevantIndex - 1) &&
+                $nextRelevantIndex - 1 < $lastNonDefaultArgumentIndex
+            ) {
+                $this->removeDefaultArgument($tokens, $nextRelevantIndex);
+            }
+            $examinedIndex = $nextRelevantIndex;
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return int|null
+     */
+    private function getLastNonDefaultArgumentIndex(Tokens $tokens, $index)
+    {
+        $nextRelevantTokenIndex = $tokens->getNextTokenOfKind($index, $this->variableOrTerminatorTokens);
+
+        if (null === $nextRelevantTokenIndex) {
+            return;
+        }
+
+        $lastNonDefaultArgumentIndex = null;
+
+        while ($tokens[$nextRelevantTokenIndex]->isGivenKind(T_VARIABLE)) {
+            if (!$tokens[$tokens->getNextMeaningfulToken($nextRelevantTokenIndex)]->equals('=') &&
+                !$this->isEllipsis($tokens, $nextRelevantTokenIndex)
+            ) {
+                $lastNonDefaultArgumentIndex = $nextRelevantTokenIndex;
+            }
+
+            $nextRelevantTokenIndex = $tokens->getNextTokenOfKind($nextRelevantTokenIndex, $this->variableOrTerminatorTokens);
+        }
+
+        return $lastNonDefaultArgumentIndex;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $variableIndex
+     *
+     * @return bool
+     */
+    private function isEllipsis(Tokens $tokens, $variableIndex)
+    {
+        if (PHP_VERSION_ID < 50600) {
+            return $tokens[$tokens->getPrevMeaningfulToken($variableIndex)]->equals('.');
```

Comment:
@keradus do we really want to go this direction? Fixing weird token sequences (php 7 code tokenized with php 5 etc.)?

IMO if we do it here, we have to do it in many many other places.

Example: `$a <=> $b`
Here we have the spaceship operator (`T_SPACESHIP`), but php 5 tokenizer returns `<=` and `>` tokens, so we have to binary operators.
Result when fixing with php 5: `$a <= > $b`, which is invalid.